doqclient.go

```diff
@@ -224,9 +223,16 @@ func (s *doqConnection) getStream() (quic.Stream, error) {
 
 	stream, err := s.connection.OpenStream()
 	if netErr, ok := err.(net.Error); ok && (netErr.Timeout() || netErr.Temporary()) {
-		// Try to open a new connection
+		// Try to open a new connection, but clean up our mess before we do so
 		_ = s.connection.CloseWithError(DOQNoError, "")
-		s.connection, err = quicDial(s.hostname, s.endpoint, s.lAddr, s.tlsConfig, s.config, s.pool)
+		// and then we need to close the connection / socket ourselves as we own
+		// the UDP socket not quic-go
+		// c.f. https://github.com/quic-go/quic-go/issues/1457
+		_ = s.udpConn.Close()
+		s.udpConn = nil
```

Comment:
Same comment as above, just make sure we don't run into a nil-ptr panic here ever.

Reply:
Yeah, will do